### PR TITLE
doc: reword README and examples

### PR DESCRIPTION
**Pull request by divarvel**
- include token authorization in all the examples. We don't want to mislead people; we need to be explicit about authorizing the token in endpoints;
- use an IETF reference link instead of a french MDN page;
- Make the provided example an actual server instead of a test (integration tests are already carried out anyway);
- Make the provided example the same as the README file (ignoring extra comments).

I'm not sure about what should be put in the doctest example. Since it's not a good idea to start a long-running http server inside tests, we have to comment out the part that actually starts it; it's a little bit awkward.
Also, it generates a keypair at startup, which is a bit strange in the context of securing a server: usually we start with an existing keypair. Maybe we can hardcode a public key.

### CODE REVIEW

File: examples/readme.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+You can also use the following:
+
+  public key: 2d6a07768e5768192870f91a6949cd09ce49865f2e2eb1241369c300ee7cc21f
+  token: EnYKDBgDIggKBggGEgIYDRIkCAASIDZy3NpVVceLLr5Xqcv08H7BeBQry38djs13jJz6uDxVGkBWQyFbDLPYaEo1PMZxB6In0mbYFiAjWEJfd2kr7P2qu8YQDNCoyIBsRP4A-4OzfvzFr2o3x9b7jOHksiRxbpILIiIKIITNLF9dFYE_tbpsqBEgno0bbwLi56dvpM43SaK7o8Iu
+
+The token has to be set in the authorization header:
+
+  curl -v http://localhost:8080/hello \
+    -H "Authorization: Bearer <token>"
+"#
```
Comment (Tipnos) [inline examples/readme.rs:27]: Is fmt responsible for 0 indent?

**Review by Tipnos — APPROVED** [review]
Nice refinement!

File: examples/readme.rs (hunk 1 above)
Comment (divarvel) [reply examples/readme.rs:27]: this is a multiline string, so leaving indents in would also indent the lines in the output.

Comment (divarvel) [reply examples/readme.rs:27]: there are external macros for that, but i don't think it's worth depending on them here.

**Pull request merged**